textfsm_test.py

```diff
@@ -610,6 +610,26 @@ def testList(self):
     result = t.ParseText(data)
     self.assertEqual(str(result), ("[[['one'], 'two']]"))
 
+
+  def testNestedMatching(self):
```

Comment:
It's also a good idea to test behaviour if group names conflict in the template.

Reply:
Yes, testing should confirm that the conflict is handled gracefully. Or not, if that is the intention.

Reply:
If the value 'name' is defined twice within the same List Value we would expect textfsm.TextFSMTemplateError to be raised. There should be a test to confirm that exception case.

Reply:
I have added better exception handling in my latest commit that raises the correct error type and preserves the specific regex error text.